spec/models/nomis/sentence_detail_spec.rb

```diff
@@ -1,53 +1,117 @@
-require 'rails_helper'
+require "rails_helper"
 
 describe Nomis::SentenceDetail, model: true do
   let(:date) { Date.new(2019, 2, 3) }
   let(:override) { Date.new(2019, 5, 3) }
 
-  describe '#automatic_release_date' do
-    context 'when override present' do
+  describe "#automatic_release_date" do
+    context "when override present" do
       before do
         subject.automatic_release_date = date
         subject.automatic_release_override_date = override
       end
 
-      it 'overrides' do
+      it "overrides" do
         expect(subject.automatic_release_date).to eq(override)
       end
     end
 
-    context 'without override' do
+    context "without override" do
       before do
         subject.automatic_release_date = date
         subject.automatic_release_override_date = nil
       end
 
-      it 'uses original' do
+      it "uses original" do
         expect(subject.automatic_release_date).to eq(date)
       end
     end
   end
 
-  describe '#conditional_release_date' do
-    context 'when override present' do
+  describe "#conditional_release_date" do
+    context "when override present" do
       before do
         subject.conditional_release_date = date
         subject.conditional_release_override_date = override
       end
 
-      it 'overrides' do
+      it "overrides" do
         expect(subject.conditional_release_date).to eq(override)
       end
     end
 
-    context 'without override' do
+    context "without override" do
       before do
         subject.conditional_release_date = date
       end
 
-      it 'uses original' do
+      it "uses original" do
         expect(subject.conditional_release_date).to eq(date)
       end
     end
   end
+
+  describe "#recall_release_date" do
```

Comment:
I like what these tests cover. I think using more cascading contexts and lets could make them easier to read, though.

I played with this a little to try and demonstrate what I mean. Happy to pair on it (we don't need to make changes in this PR, necessarily!):
```ruby
describe "#recall_release_date" do
  before do
    subject.actual_parole_date = parole_date
    subject.post_recall_release_date = post_recall_release_date
    subject.post_recall_release_override_date = override_date
  end

  context 'when there are no dates' do
    let(:parole_date) { nil }
    let(:post_recall_release_date) { nil }
    let(:override_date) { nil }

    it 'returns nil' do
      expect(subject.recall_release_date).to be_nil
    end
  end

  context 'when there is no post-recall release date' do
    let(:post_recall_release_date) { nil }

    context 'and a parole date' do
      let(:parole_date) { 12.weeks.from_now }

      context 'and no post-recall override date' do
        let(:override_date) { nil }

        it 'returns the parole date' do
          expect(subject.recall_release_date).to eq(parole_date)
        end
      end

      context 'and a later post-recall override date' do
        let(:override_date) { parole_date + 4.weeks }

        it 'returns the parole date' do
          expect(subject.recall_release_date).to eq(parole_date)
        end
      end

      context 'and an earlier post-recall override date' do
        let(:override_date) { parole_date - 4.weeks }

        it 'returns the override date' do
          expect(subject.recall_release_date).to eq(override_date)
        end
      end
    end
  end
end
```

Reply:
(note that first test fails, exposing an error I've talked about in another comment)